### PR TITLE
ide: fix issue with large documents

### LINKED ISSUE #1823: interpreter crashes after server boot when very large scd file is currently open
I am not sure what is wrong with my system, but all builds, including the downloaded 3.7 binary crash the interpreter on server start. I've cleaned/rebuilt sc serveral times, and rebooted the system.

```
Time Awake Since Boot: 2600 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_PROTECTION_FAILURE at 0x00007fff58ff5a98
Exception Note:        EXC_CORPSE_NOTIFY

VM Regions Near 0x7fff58ff5a98:
    MALLOC_TINY            00007f8744800000-00007f8744900000 [ 1024K] rw-/rwx SM=PRV  
--> STACK GUARD            00007fff558d5000-00007fff590d5000 [ 56.0M] ---/rwx SM=NUL  stack guard for thread 0
    Stack                  00007fff590d5000-00007fff598d5000 [ 8192K] rw-/rwx SM=PRV  thread 0

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   sclang                        	0x0000000106469f46 ScIDE_SetDocTextMirror(VMGlobals*, int) + 150
1   sclang                        	0x0000000106344d41 doPrimitive(VMGlobals*, PyrMethod*, int) + 209
2   sclang                        	0x0000000106386b0d Interpret(VMGlobals*) + 24525
3   sclang                        	0x0000000106380964 runInterpreter + 36
4   sclang                        	0x000000010638db40 runLibrary + 80
5   sclang                        	0x00000001063be1f1 SC_LanguageClient::runLibrary(PyrSymbol*) + 33
6   sclang                        	0x00000001063c396a SC_TerminalClient::interpretInput() + 154
7   sclang                        	0x00000001063c7d9e void boost::asio::asio_handler_invoke<boost::_bi::bind_t<void, boost::_mfi::mf0<void, SC_TerminalClient>, boost::_bi::list1<boost::_bi::value<SC_TerminalClient*> > > >(boost::_bi::bind_t<void, boost::_mfi::mf0<void, SC_TerminalClient>, boost::_bi::list1<boost::_bi::value<SC_TerminalClient*> > >&, ...) + 110
8   sclang                        	0x00000001063c7cd4 boost::asio::detail::completion_handler<boost::_bi::bind_t<void, boost::_mfi::mf0<void, SC_TerminalClient>, boost::_bi::list1<boost::_bi::value<SC_TerminalClient*> > > >::do_complete(boost::asio::detail::task_io_service*, boost::asio::detail::task_io_service_operation*, boost::system::error_code const&, unsigned long) + 84
```

### PR DESCRIPTION
**Pull request by miguel-negrao**
fixes #1823

### CODE REVIEW

**Comment by miguel-negrao** [comment]
Would malloc/free also work on windows ? If so the ifdefs could be removed. Sorry for the newbiness...

**Comment by timblechmann** [comment]
lgtm. yes,  the windows ifdef should go away. alloca and variable length arrays are almost doing the same thing and are both prone to stack overflows

**Comment by miguel-negrao** [comment]
Ok, removed ifdef.

File: editors/sc-ide/primitives/sc_ipc_client.cpp [hunk 1]
```diff
@@ -384,11 +384,8 @@ int ScIDE_SetDocTextMirror(struct VMGlobals *g, int numArgsPushed)
...
     
-#ifndef _WIN32
-    char text[length + 1];
-#else
-    char * text = (char*)_alloca(length + 1);
-#endif
+    char * text = (char*)malloc(length + 1);
+    
     if (slotStrVal( textSlot, text, length + 1))
         return errWrongType;
```
Comment (timblechmann) [inline editors/sc-ide/primitives/sc_ipc_client.cpp]: memory leak

**Comment by miguel-negrao** [comment]
Sorry, I missed those cases.

Ok, tried using vector instead. I was misunderstading vector regarding memory allocation. I though vector allocates on the stack, after investigating more my understanding now is that by default vector allocates the actual objects (is this case chars) on the heap, and the pointer is allocated on the stack, then when the vector goes out of scope it will release the memory of the underlying objects automatically. Is my understanding correct ?

Does this commit look good ?
https://github.com/miguel-negrao/supercollider/commit/044095abe316ba5fd59d968d095d1b451b71f53f

**Comment by timblechmann** [comment]
`std::vector` does all the underlying memory-management for you. c++11 also has `std::array`, which allocates a fixed-sized array on the stack.

not sure about the requirements/guarantees of `execve`

**Comment by miguel-negrao** [comment]
Ok, I've opened another pull request hopefully fixing the memory leak.

**Comment by bagong** [comment]
Possibly, Miguel, you made a major contribution to stabilizing the Windows build with this fix. I am kept off trying out the Windows build at leisure by other stuff in my todo-list, but I noticed that my recent builds are markedly more stable than the ones I used before I integrated your fix. I hope to be ready tomorrow and see how it ultimately goes.

**Pull request merged**